src/inputType/PinchInput.ts

```diff
@@ -203,17 +104,169 @@ export class PinchInput implements IInputType {
 	 * @method eg.Axes.PinchInput#disable
 	 * @return {eg.Axes.PinchInput} An instance of a module itself <ko>모듈 자신의 인스턴스</ko>
 	 */
-	disable() {
-		this.hammer && (this.hammer.get("pinch").options.enable = false);
+	public disable() {
+		this.isEnabled = false;
 		return this;
 	}
+
 	/**
 	 * Returns whether to use an input device
 	 * @ko 입력 장치를 사용 여부를 반환한다.
 	 * @method eg.Axes.PinchInput#isEnable
 	 * @return {Boolean} Whether to use an input device <ko>입력장치 사용여부</ko>
 	 */
-	isEnable() {
-		return !!(this.hammer && this.hammer.get("pinch").options.enable);
+	public isEnable() {
+		return this.isEnabled;
+	}
+
+	private onPinchStart(event: InputEventType) {
+		if (event instanceof PointerEvent) {
+			this.addPointerEvent(event);
+		}
+		if (!this.isEnable() || this.getTouches(event) !== 2) {
+			return;
+		}
+
+		this.baseValue = this.observer.get(this)[this.axes[0]];
+		this.firstTouch = event instanceof TouchEvent ? event : null;
+		this.observer.hold(this, event);
+		this.pinchFlag = true;
+		const pinchEvent = this.transformEvent(event);
+		this.prevInput = pinchEvent;
+	}
+
+	private onPinchMove(event: InputEventType) {
+		if (event instanceof PointerEvent) {
+			this.addPointerEvent(event);
+		}
+		if (!this.pinchFlag || !this.isEnable() || this.getTouches(event) !== 2) {
+			return;
+		}
+
+		const pinchEvent = this.transformEvent(event);
+		const offset = this.getOffset(pinchEvent.scale, this.prevInput.scale);
+		this.observer.change(this, event, toAxis(this.axes, [offset]));
+		this.prevInput = pinchEvent;
+	}
+
+	private onPinchEnd(event: InputEventType) {
+		if (event instanceof PointerEvent) {
+			this.removePointerEvent(event);
+		}
+		if (!this.pinchFlag || !this.isEnable() || this.getTouches(event) > 2) {
+			return;
+		}
+
+		this.observer.release(this, event, toAxis(this.axes, [0]), 0);
+		this.baseValue = null;
+		this.pinchFlag = false;
+		this.firstTouch = null;
+		this.prevInput = null;
+	}
+
+	private transformEvent(event: InputEventType): PinchEvent {
+		if (event instanceof PointerEvent) {
+			return {
+				srcEvent: event,
+				scale: this.getScaleFromPointers(),
+			};
+		} else if (event instanceof TouchEvent) {
+			return {
+				srcEvent: event,
+				scale: this.getScaleFromTouch(this.firstTouch.touches, event.touches),
+			};
+		}
+	}
+
+	private attachEvent(observer: IInputTypeObserver) {
+		this.observer = observer;
+		this.isEnabled = true;
+		this.activeInput = convertInputType(this.options.inputType);
+		if (this.activeInput === "pointer") {
```

Comment:
The code is not perfect, but it would be good to see it as pseudo code to understand the context. The code right now is not bad, but I think it would be good to think about a more readable code.
```js
//////////////////////////////////////////////////////
const hasPointEvent = "PointerEvent" in window;
const hasMSPointEvent = "MSPointerEvent" in window;
const SUPPORT_POINTER_EVENTS = hasPointEvent|| hasMSPointEvent;
const SUPPORT_TOUCH = "ontouchstart" in window;
const pointEventNames = hasPointEvent?["pointerdown","pointermove","pointerup","pointercancel"]:["MSPointerDown","MSPointerMove","MSPointerUp","MSPointerCancel"];


function getEventNames(inputType){
    if(inputType === "pointer"){
        return pointEventNames;
    }else{
        return ["touchstart", "touchmove", "touchend", "touchcancel"];
    }
}
//////////////////////////////////////////////////////

this.activeInput = convertInputType(this.options.inputType);
const [POINTER_DOWN, POINTER_MOVE, POINTER_UP, POINTER_CANCAL] = getEventNames(this.activeInput);

this.element.addEventListener(POINTER_DOWN, this.onPinchStart, false);
this.element.addEventListener(POINTER_MOVE, this.onPinchMove, false);
this.element.addEventListener(POINTER_UP, this.onPinchEnd, false);
this.element.addEventListener(POINTER_CANCAL, this.onPinchEnd, false);
```

Reply:
기존 hammerjs의 TouchMouseInput 및 새로 구현된 형태에서 activeInput이 "touchmouse"로 설정된 경우처럼 POINTER_DOWN 이벤트의 종류가 여러 종류일 수 있으므로 하나의 상수 POINTER_DOWN를 addEventListener에 사용하지는 못합니다.
그러므로 최근 커밋을 통해 convertInputType에서 activeInput에 start, move, end 콜백을 호출하는 이벤트 종류로 분류된 오브젝트를 리턴하는 형태로 개선이 이루어졌습니다.

There can be multiple events for POINTER_DOWN like TouchMouseInput from legacy hammerjs. as "touchmouse" for activeInput in new code. So we cannot use const POINTER_DOWN for addEventListener.
Therefore, this part has refactored which uses an object in recent commit. returning objects classified by event types that call start, move, and end callbacks.

https://github.com/naver/egjs-axes/pull/165/commits/2993a1aef4bb0d5780a48b3453977941f286005d